@here/harp-mapview/lib/MapView.ts

```diff
@@ -2620,6 +2660,92 @@ export class MapView extends THREE.EventDispatcher {
         this.m_env.entries.$frameNumber = this.m_frameNumber;
     }
 
+    /**
+     * Update the directional light camera. Note, this requires the cameras to first be updated.
+     */
+    private updateLights() {
+        // TODO: HARP-9479Globe doesn't support shadows.
+        if (
+            !this.shadowsEnabled ||
+            this.directionalLight === undefined ||
+            this.projection.type === ProjectionType.Spherical
+        ) {
+            return;
+        }
+
+        const NDCToView = (vector: Vector3Like): THREE.Vector3 => {
+            return (
+                new THREE.Vector3(vector.x, vector.y, vector.z)
+                    .applyMatrix4(this.camera.projectionMatrixInverse)
+                    // Make sure to apply rotation, hence use the rte camera
+                    .applyMatrix4(this.m_rteCamera.matrixWorld)
+            );
+        };
+        const ViewToLightSpace = (worldPos: THREE.Vector3, camera: THREE.Camera): THREE.Vector3 => {
+            return worldPos.applyMatrix4(camera.matrixWorldInverse);
```

Comment:
Is it expected that `worldPos` is modified?

Reply:
Yes, I will add comment.